src/pages/AddressesPage.js

```diff
@@ -8,24 +8,31 @@ import Loader from "components/Loader";
 import { AuthContext } from "components/auth/Auth";
 import Card from "components/Card";
 import AddressTable from "components/addresses/AddressTable";
+import AddressGroupTable from "components/addresses/AddressGroupTable";
 import IconButton from "components/IconButton";
 import useSelectedRow from "hooks/useSelectedRow";
 import "./Dashboard.scss";
 
 const AddressesPage = () => {
-  const { user, skipKYC } = useContext(AuthContext);
+  const { user } = useContext(AuthContext);
   const history = useHistory();
   const getAddressesUrl = `/user/${user.id}/address`;
-  const [selected, setSelected, selectRowConfig] = useSelectedRow(null);
+  const [selected, , selectRowConfig] = useSelectedRow(null);
+  const [selectedGroup, , selectGroupRowConfig] = useSelectedRow(null);
   const { data: addresses, error: fetchAddressError } = useSWR(getAddressesUrl);
   const isFetchingAddresses = !addresses && !fetchAddressError;
   const hasMultipleAddresses = addresses?.length > 1;
-
+  const [unGroupAddress, setUnGroupAddress] = useState([]);
+  const [groupAddress, setGroupAddress] = useState([]);
+  const { data: settings, error: fetchSettingsError } = useSWR(
+    `/settings/${user.id}`
+  );
   useEffect(() => {
-    if (!addresses) return;
-    const hasAddress = addresses.find((a) => a.id === selected);
-    if (!hasAddress) setSelected(null);
-  }, [addresses, selected, setSelected]);
+    const groupAddress = addresses?.filter((i) => i.groupID);
```

Comment:
you dont need to do this logic in useEffect. It can be done on each render right?